tmt/utils/__init__.py

```diff
@@ -6347,3 +6351,31 @@ def is_url(url: str) -> bool:
     """ Check if the given string is a valid URL """
     parsed = urllib.parse.urlparse(url)
     return bool(parsed.scheme and parsed.netloc)
+
+
+@contextlib.contextmanager
+def suppress_warning(category: Optional[type[Warning]]) -> Iterator[None]:
+    """
+    Optionally disable the given warning.
+
+    Using this context manager you can suppress a warning. This warning gets re-enabled with an
+    exit from this context manager.
+
+    The example can be suppressing of the urllib insecure request warning:
+
+    .. code-block:: python
+
+        with suppress_warning(urllib3.exceptions.InsecureRequestWarning):
+            ...
+
+    """
+    if category is not None:
+        with _suppress_warning_lock:
+            try:
+                warnings.simplefilter('ignore', category)
```

Comment:
Not saying this is wrong, just wonder if it wouldn't be possible/better to shoehorn the catch_warnings context manager https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings ?

Reply:
The `warnings.catch_warnings` approach seems cleaner and more universal. What do you think? Could you please review?

- https://github.com/teemtee/tmt/pull/3356/commits/13eb123d9a762d8128f4cddd232059496c3aacf2

@happz Could you please also take a look? Thanks!

Reply:
@seberm awesome :)  

Wouldn't it make sense to use a different name? 

fwiw looking at warnings code: `if action == "all": return "always" # Alias` .. I'd probably remove the 'all' from `ActionType`
They also mention: "action: error, ignore, always, default, module, or once"

Out of curiosity, what would `ruff format` prefer:
```
with _catch_warning_lock, warnings.catch_warnings():
```
or
```
with _catch_warning_lock:
    with warnings.catch_warnings():
```
? Or is it ok with both? (can check myself later since I'm the one curious 🐕‍🦺)

Reply:
LGTM, one comment WRT naming, but that's all I have.

Reply:
Hello,
- Removed the `all` alias from the `ActionType` (https://github.com/teemtee/tmt/pull/3356/commits/417272be4973ae4a84127497ca2c5132461c8166).
- Renamed the `utils.catch_warnings` to `utils.catch_warnings_safe` (https://github.com/teemtee/tmt/pull/3356/commits/61b34f3fe363f1089512d8972c4d58942dd5ca42). Other names could be the following (no strong opinion there):
    - `catch_warnings_threadsafe`
    - `thread_safe_catch_warnings`
    - `catch_warnings_mt` ("multithreaded")
    - `catch_warnings_mt_safe`

> Out of curiosity, what would ruff format prefer:
> ...

Ruff prefers one `with` statement with multiple context managers separated by a comma:
- https://docs.astral.sh/ruff/rules/multiple-with-statements/